ideas/167-paid-master-node.md

```diff
@@ -0,0 +1,187 @@
+- QA: TBD
+- Evaluation: TBD
+- Clojure Contributor: TBD
+- Go Contributor: TBD
+- PM: TBD
+- UX(R): TBD
+- Community: @andytudhope
+
+## Product Overview
+
+This idea takes of where https://github.com/status-im/ideas/blob/master/ideas/1-offline-inboxing.md ended. It provides an SNT payment layer for usage of master nodes.
+
+> We want to make it easy to run a Whisper mail server (part of Whisper v5 API). This will archive messages and forward them to a peer upon request. You can run this yourself if you want to, but we’ll also provide default mail servers through our clusters, and make it easy for anyone to set one up. This can be paid using SNT, which aligns incentives in terms of Quality of Service.
+
+We need something users can do with their SNT, to realize the power of the
+technology and keep them interested.
+
+Currently offline inboxing is a free resource, subsidised and run by us. This is
+sub-optimal for many reasons, and goes against our vision of letting users pay
+for things they use, as well as incentivizing this capability to be
+decentralized.
+
+Offline inboxing is run on mail server aka master nodes. In the scope of e.g.
+86-push-notifications push notifications will likely be served through that as
+well.
+
+### Constraints / objections:
+
+- No one will pay for it.
+
+This is a free resource. If we want to subsidise it and the concern is people
+  not wanting to stay / top up to use basic funcrtionality we can solve it by
+  giving free SNT as part of SNT onboarding.
+  
+- Free SNT?!
+
+Yes. For a closed beta limited to 40k people, we can limit the free amount to 5
+SNT per person, which limits the downside risk this to 200k SNT ($20k at the
+time of this writing).
+
+- What about fraud?
+
+We can use already collected emails as identity and dedup by it. Max downside
+per email (pre-signed up) is 5 SNT, and to get it out would still be a
+transaction, as well as a signal to us.
+
+- How does this scale?
+
+For open beta we can figure out how to deal with SNT onboarding by e.g. using
+trustlines.
+
+---
+
+For more on rationale of SNT utility, see the whitepaper. Providing onboarding
+SNT also furthers distribution of tokens as well as transactions on their own, a
+good.
+
+This is a hypothesis, but this is the whole basis of Status.
+
+### User stories
+
+User story 1: Bobby Blockchain runs a master node and gets paid by peers in SNT for services provided.
+
+User story 2: Alice pays SNT to a master node in order to get offline inboxing for some period of time.
+
+Out of scope for this idea:
+
+- (User story 3: Decentralized discovery mechanism for master nodes)
+
+- (User story 4: Onboarding mechanism to seed closed beta users with SNT)
+
+User story 4 is sufficiently different in focus to warrant a separate swarm.
+Since it might be a pre-requsite from a UX point of view, the behavior in user
+story 1 and 2 should be toggle-able with a flag. I.e. we make sure we can still
+provide free service for the time being.
```

Comment:
How will we encourage users to try this if it remains available for free? Agreed a soft launch is better but perhaps requires a beta user campaign.